### DISCUSSION TITLE
CSSBattle Previewer 2.0

### DISCUSSION THREAD

**Discussion by Beowolve** [cssbattle/forum#258] [Show and tell]
Since the last backend update a new chromium is used on the backend, so I decided to fork the cssbattle previewer and setup a new backend on firebase for it.
Some stuff didn't work on codepen, so I moved it to codesandbox where everything worked as intended. Also if the project gets bigger, this might be a better place to host it anyway.

Here is the link to the current version:
https://tc70f3.csb.app/
or
https://tc70f3.codesandbox.io/

I have added some features and fixed some bugs from the original code in the process. Hopefully not added to many new bugs ;).

![image](https://user-images.githubusercontent.com/32326640/167362839-f9113faf-a30a-4df6-b93c-82ae45d3b175.png)

Have fun,
Beo

**Comment by APerricone** [comment]
This piece of code takes 100% on [#3](https://cssbattle.dev/play/3)
```html
<p><style>*{background:#6592CF}*+*{margin:75+50+75;background:#243D83;padding:50+0}p{margin:0+125;height:50;background:#EEB850;border-radius:50%;box-shadow:0+0+0+50px#243D83,0+0+0+100px#6592CF
```
but the result is pretty different
![image](https://user-images.githubusercontent.com/22457168/166248435-27e99473-bd2f-4ac2-9c56-46e3e2610b1f.png)
very weird

**Reply by APerricone** [reply]
😲 I updated chrome and now it renders as the preview 😲 the box-shadow does not ingrease the radius

**Reply by Beowolve** [reply]
Yeah, it looks like cssbattle backend is still a little behind the current browser.
But this must be a bug, this would break so many things? Does anybody know if this is an intended change?

**Reply by rafehimself** [reply]
they actualize browser at least once a month, I'm not sure will we be able to catch up,
I've just check and there is a new update 🤣

**Reply by Beowolve** [reply]
Well the previewer only needs to catch up with the cssbattle backend, not with the chromium releases so in this case I will need to downgrade the backend version. I take a look it later.
Working on the regular website now also means you will need to use an older chromium version at the moment, or you have the same problem.

For everyone interrested, here you can see all the embedded chromium versions from puppeteer:
https://github.com/puppeteer/puppeteer/blob/main/versions.js

**Reply by APerricone** [reply]
I will never understand why they don't say which version is used, for example on #248

**Reply by Beowolve** [reply]
I have downgraded Puppeteer to Version 13.5.2, this is the last version where u get correct circles with box-shadow.
